tempesta_fw/http_parser.c

```diff
@@ -1842,21 +1871,21 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len)
 		if (unlikely(!IN_ALPHABET(c, hdr_a)))
 			return TFW_BLOCK;
 
-		tfw_http_msg_hdr_open(msg, p);
+		__msg_hdr_open(p);
 
 		switch (LC(c)) {
 		case 'c':
 			__FSM_MOVE(Req_HdrC);
 		case 'h':
-			if (likely(p + 5 <= data + len
+			if (likely(__data_available(p, 5)
```

Comment:
Nope. There would to much copy-paste here in that case as token length used no only in `__data_available` . I think we shouldn't mix all the possible enhancements in one patch-set.